app/sequencescape/sequencescape/api/v2/shared/has_workline_identifier.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Sequencescape::Api::V2::Shared
+  # Include in API endpoints that have barcodes to add a few standard methods
+  module HasWorklineIdentifier
+    extend ActiveSupport::Concern
+
+    # Finds the labware whose barcode will be printed in the label at the top_right field.
+    # Uses alternative_workline_identifier from the purpose config if present, otherwise uses the stock plate.
+    def workline_reference
+      alternative_workline_identifier_purpose = SearchHelper.alternative_workline_reference_name(self)
+      return stock_plate if alternative_workline_identifier_purpose.nil?
+
+      ancestors.where(purpose_name: alternative_workline_identifier_purpose).last
```

Comment:
I wonder if this would be easier for someone to follow if we avoided the inline if on the first line and made it clear that the other line would be returned in the else case?

```suggestion
      if alternative_workline_identifier_purpose.nil?
        return stock_plate
      else
        return ancestors.where(purpose_name: alternative_workline_identifier_purpose).last
      end
```

I think you can even drop the `return` key words then, but that leaves scope for more code being added below this block and creating an unexpected bug.

Reply:
Thanks, I think I agree it's clearer but Rubocop doesn't agree - https://docs.rubocop.org/rubocop/cops_style.html#styleguardclause

So I think I'll stick with how it is in the interests of keeping the code style consistent.

Reply:
Maybe it can't tell the difference?